src/api/data/cache/cache.go

```diff
@@ -148,37 +148,84 @@ func (c *cachedCharts) Refresh() error {
 		if err != nil {
 			return err
 		}
+
+		// 2 - Parse repo index
 		charts, err := helpers.ParseYAMLRepo(body, repo.Name)
 		if err != nil {
 			return err
 		}
+
+		// 3 - Process elements in index
 		var chartsWithData []*models.ChartPackage
+		// Buffered channel
+		ch := make(chan chanItem, len(charts))
+		// 3.1 - parallellize processing
 		for _, chart := range charts {
-			// Extra files. Skipped if the directory exists
-			dataExists, err := charthelper.ChartDataExists(chart)
-			if err != nil {
-				return err
-			}
-			if !dataExists {
-				log.WithFields(log.Fields{
-					"name":    *chart.Name,
-					"version": *chart.Version,
-				}).Info("Local cache missing")
-
-				err := charthelper.DownloadAndExtractChartTarball(chart)
-				if err != nil {
-					log.WithFields(log.Fields{
-						"error": err,
-					}).Error("Error on DownloadAndExtractChartTarball")
-					// Skip chart if error extracting the tarball
-					continue
-				}
-				// If we have a problem processing an image it will fallback to the default one
-				charthelper.DownloadAndProcessChartIcon(chart)
+			go processChartMetadata(chart, ch)
+		}
+		// 3.2 Channel drain
+		for range charts {
+			it := <-ch
+			// Only append the ones that have not failed
+			if it.err == nil {
+				chartsWithData = append(chartsWithData, it.chart)
 			}
-			chartsWithData = append(chartsWithData, chart)
 		}
-		c.allCharts[repo.Name] = chartsWithData
+		updatedCharts[repo.Name] = chartsWithData
 	}
+
+	// 4 - Update the stored cache with the new elements if everything went well
+	c.rwm.Lock()
+	c.allCharts = updatedCharts
+	defer c.rwm.Unlock()
 	return nil
 }
+
+// Represents every element processed in paralell
+type chanItem struct {
+	chart *models.ChartPackage
+	err   error
+}
+
+// Counting semaphore, 25 downloads max in paralell
+var tokens = make(chan struct{}, 25)
+
+func processChartMetadata(chart *models.ChartPackage, out chan chanItem) {
```

Comment:
Because out is only used as a send channel, you can change the 2nd parameter specification to `out chan<- chanItem`.